### DISCUSSION TITLE
[suggestion]  Add ctor to ReseEase that allow to add both RequestModifier and HttpMessageHandler

### DISCUSSION THREAD

**Discussion by Legion-ale** [canton7/RestEase#231] [General]
As explained in the title i'm in the following situation:
```
private readonly Microsoft.Extensions.Http.PolicyHttpMessageHandler retryPolicyHandler = new Microsoft.Extensions.Http.PolicyHttpMessageHandler(
	Policy.HandleResult<HttpResponseMessage>(r => r.StatusCode == System.Net.HttpStatusCode.BadGateway)
			.RetryAsync()
);

private readonly RequestModifier proxyHandler = async (request, cancellationToken) =>
{
	//Lot of code to set some proxy stuff before sending the call
};
```
But when i try to inizialize RestEase i cannot add both to the ctor (even if i use new RestClient(...)) and i need to choose only one of the 2 following ctors:
`RestClient.For<IWebApi>(API_URL, proxyHandler);` (ctor that accept RequestModifier)
`RestClient.For<IWebApi>(API_URL, retryPolicyHandler);` (ctor that accept HttpMessageHandler)
There are not public members that allow me to edit that configuration in a second time because HttpClient is a private property (as it should be).

Thanks and sorry for my bad english :)
L-

**Comment by canton7** [comment]
`RequestModifier` is implemented using a `HttpMessageHandler`: [ModifyingClientHandler](https://github.com/canton7/RestEase/blob/2922d6d8d3cb184bd8ef4d54c71251c03e55d5cd/src/RestEase/Implementation/ModifyingClientHttpHandler.cs), but until recently it wasn't a `DelegatingHandler`, so this wouldn't have been possible.

However, `ModifyingClientHandler` was recently changed to inherit from `DelegatingHandler`, so this is now possible. I'm slightly reluctant to implement it though, as it will increase the number of constructors quite significantly, as it's already quite messy!

As a workaround, you should be able to do something like:

```cs
var clientHandler = ModifyingClientHandler(proxyHandler) { InnerHandler = retryPolicyHandler };
var client = RestClient.For<T>(clientHandler);
```

**Comment by Legion-ale** [comment]
Many thanks, i didn't know the `ModifyingClientHandler` class, with that I think I can solve my problem, tomorrow i'll test it :)
Thaks again for all the effort,
L-

***EDIT*** All works as expected, now my code is like that:

```
private static readonly Microsoft.Extensions.Http.PolicyHttpMessageHandler retryPolicyHandler = new Microsoft.Extensions.Http.PolicyHttpMessageHandler(
	Policy.HandleResult<HttpResponseMessage>(r => r.StatusCode == System.Net.HttpStatusCode.BadGateway).WaitAndRetryAsync(retryCount: 3, retry => TimeSpan.FromSeconds(retry * 10))
) { InnerHandler = new HttpClientHandler() };

private static readonly RequestModifier proxyHandler = async (request, cancellationToken) =>
{
	//Lot of code to set some proxy stuff before sending the call
};

private static readonly RestEase.Implementation.ModifyingClientHttpHandler clientHandler = new RestEase.Implementation.ModifyingClientHttpHandler(proxyHandler) { InnerHandler = retryPolicyHandler };

public MyService(....)
{
	_api = RestClient.For<IWebApi>(API_URL, clientHandler);
}
```
